src/apps/emess.cpp

```diff
@@ -27,8 +27,19 @@ emess(int code, const char *fmt, ...) {
 	va_start(args, fmt);
 	/* prefix program name, if given */
 	if (emess_dat.Prog_name != nullptr)
-		(void)fprintf(stderr,"%s\n<%s>: ",pj_get_release(),
-                              emess_dat.Prog_name);
+    {
+        // For unit test purposes, allow PROJ_DISPLAY_PROGRAM_NAME=NO
+        const char* pszDisplayProgramName = getenv("PROJ_DISPLAY_PROGRAM_NAME");
+        if( pszDisplayProgramName && strcmp(pszDisplayProgramName, "NO") == 0 )
+        {
+            // do nothing
+        }
+        else
+        {
+            (void)fprintf(stderr,"%s\n<%s>: ",pj_get_release(),
+                                  emess_dat.Prog_name);
+        }
```

Comment:
Is there a reason this expression can't be negated? Something like 
```suggestion
        if( !pszDisplayProgramName && strcmp(pszDisplayProgramName, "NO") != 0 )
        {
            (void)fprintf(stderr,"%s\n<%s>: ",pj_get_release(),
                                  emess_dat.Prog_name);
        }
```

Reply:
> Is there a reason this expression can't be negated ?

no, just that humain brain tends to have issues with negation of clauses with and operation. Your negation proposal was wrong ;-).
I've changed to ``if( !(pszDisplayProgramName && strcmp(pszDisplayProgramName, "NO") == 0) )``

Reply:
> Your negation proposal was wrong ;-). 

Hence the "something like" :-) Improved readibility is a valid reason for your original code. Just looked a bit odd having a code branch that does nothing

Reply:
Yet another reason to import the perl "unless" (negated if) keyword into C/C++ 🙂